addons/product/models/product_product.py

```diff
@@ -589,9 +589,10 @@ def name_search(self, name='', args=None, operator='ilike', limit=100):
                 # OR operator (and given the fact that the 'name' lookup results come from the ir.translation table
                 # Performing a quick memory merge of ids in Python will give much better performance
                 products = self.search_fetch(expression.AND([domain, [('default_code', operator, name)]]), ['display_name'], limit=limit)
+                products_query = self._search(expression.AND([domain, [('default_code', operator, name)]]), limit=limit)
```

Comment:
Oh and since it's not used unless you enter the `if`, you could move this variable inside the if :eyes: